### PR TITLE
Enhancements for HardwareSerial (Arduino Mega) and ask() with empty s…

### PR DESCRIPTION
**Pull request by ecm-bitflipper**
Based on MOVI API 1.03, these changes allow for construction of MOVI()
using a HardwareSerial pointer.  Also included is a small change to
prevent ask() from calling say() when the parameter is an empty string.

### CODE REVIEW

File: MOVIShield.cpp [hunk 1]
```diff
@@ -45,11 +45,13 @@
 
 MOVI::MOVI()
 {
+    usehardwareserial=false;
     construct(ARDUINO_RX_PIN, ARDUINO_TX_PIN, false);
 }
 
-MOVI::MOVI(bool debugonoff)
+MOVI::MOVI(uint8_t debugonoff)
```
Comment (fractor) [inline MOVIShield.cpp]: Why is this changed to uint8_t from bool?

Comment (ecm-bitflipper) [inline MOVIShield.cpp]: I mention this in the forum post I made about this.  With debugonoff defined as bool, the compiler (Arduino IDE 1.6.7) was accepting &Serial1 to that constructor without a cast even.  This was how it was behaving before I added the extra constructor for HardwareSerial \* anyway.  Perhaps we could put debugonoff back to bool and with the explicitly defined HardwareSerial \* constructor the compiler would choose the right one?

Comment (ecm-bitflipper) [inline MOVIShield.cpp]: Based on some quick research, it looks like implicit conversion from a pointer to bool is allowed and actually intended for uses such as:

if (p) {
...
}

Where the expression p returns false if p is NULL and true otherwise.

I imagine the compiler will choose the HardwareSerial \* constructor for any pointer that is a HardwareSerial instance or derivative and the bool constructor for any other pointer but I haven't tested it.  I can try it if you'd rather keep bool instead of protecting from construction with arbitrary pointers using uint8_t.  Let me know.

Comment (fractor) [inline MOVIShield.cpp]: That makes sense. It can be solved by requiring the debug parameter with the constructor that takes the hardwareserial. This makes it more consistent too:
MOVI(bool debugonoff, int rx, int tx)
MOVI(bool debugonoff, HardwareSerial *hs)

Comment (ecm-bitflipper) [inline MOVIShield.cpp]: Sure, that way is totally fine - those of us needing HardwareSerial just have to pass one additional parameter that may well always be hardcoded to false, but that's easy enough to live with.

Would you like me to submit an amended pull request?

Comment (fractor) [inline MOVIShield.cpp]: ...while you are at it. Otherwise, I'll just do it next time I am working on the library.

In anyways: Thank you very much for your contribution!

**Comment by ecm-bitflipper** [comment]
Okay, changes made as requested.  I went one further and allowed for use of the HardwareSerial \* constructor on all supported platforms, not just AVR.  It's a relatively simple change that unlocks Intel and SAM users from Serial1 if they so desire.

**Comment by fractor** [comment]
Pull request merged in. Thank you!
